### PR TITLE
Reverse Proxy Port Mapping Fixes

### PR DESCRIPTION
**Pull request by girisairam450**
If port passed along with the service then the publishing will happen based on the internal to external mapping. It is applicaple for SN API PORT, SN P2P PORT, and SL FS PORT.

### CODE REVIEW

File: scripts/bin/run-sl [hunk 1]
```diff
@@ -200,8 +200,26 @@ onTrainEnd()
         appendArrayVar "" dockerOpts "--publish=${slFsPort}:${internalFSPort}"
     fi
 
-    [[ -n "${slFsService}" ]] &&    \
+    if [[ -n "${slFsService}" ]]
+    then
         appendArrayVar "" dockerOpts "--env=SL_FS_SERVICE=${slFsService}"
+        
+        # This below logic extracts and publishes the port passed along with the
+        # SL FS service. If no port passed, it by default publishes 30303 port.
```
Comment (girisairam450) [inline scripts/bin/run-sl:208]: 30305

File: scripts/bin/run-sl [hunk 2]
```diff
@@ -200,8 +200,26 @@ onTrainEnd()
...
+        # stored inside a array variable. 
+        # Case 1 : If there is no port mentioned then the array size will be 1,
+        # Hence the default port gets published.
+        # Case 2 : Array size will be 2 when port mentioned in the service name.
+        # Hence the port number passed along with service gets published.
+        IFS=':' read -ra fsServiceArr <<< ${slFsService}
+        case "${#fsServiceArr[@]}" in
+            1) appendArrayVar "" dockerOpts "--publish=${internalFSPort}:${internalFSPort}";;
+
+            2) fsServicePort=${fsServiceArr[1]}
```
Comment (girisairam450) [inline scripts/bin/run-sl:219]: Verify if more :'s present in the service string

File: examples/reverse-proxy/common/Bind-Dockerfile [hunk 3]
```diff
@@ -8,13 +8,13 @@ FROM ubuntu:22.04
...
 RUN apt-get update                              \
  && apt-get install                             \
-    bind9=1:9.18.12-0ubuntu0.22.04.1            \
-    bind9-dnsutils=1:9.18.12-0ubuntu0.22.04.1   \
-    ca-certificates=20211016ubuntu0.22.04.1     \
-    curl=7.81.0-1ubuntu1.10                     \
-    gnupg2=2.2.27-3ubuntu2.1                    \
-    lsb-release=11.1.0ubuntu4                   \
-    ubuntu-keyring=2021.03.26                   \
+    bind9                                       \
```
Comment (girisairam450) [inline examples/reverse-proxy/common/Bind-Dockerfile:11]: Checkin with separate commit with desc

**Review by girisairam450 — COMMENTED** [review]
Reviewed

File: scripts/bin/run-sl (hunk 1 above)
Comment (girisairam450) [reply scripts/bin/run-sl]: Done

File: examples/reverse-proxy/common/Bind-Dockerfile (hunk 3 above)
Comment (girisairam450) [reply examples/reverse-proxy/common/Bind-Dockerfile:11]: Done

File: scripts/bin/run-sl (hunk 2 above)
Comment (girisairam450) [reply scripts/bin/run-sl:219]: Handled
Example
$ echo "sl.fs.service:fs:30305" | { IFS=':' read -ra line; echo ${line[0]} ${line[-1]} ; }
sl.fs.service 30305
$ echo "sl.fs.service:30305" | { IFS=':' read -ra line; echo ${line[0]} ${line[-1]} ; }
sl.fs.service 30305

**Review by suresh-ls — APPROVED** [review]

**Review by RadhakrishnaJ — APPROVED** [review]

**Pull request merged**